### PR TITLE
Cleanup non-wrapper references

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java [hunk 1]
```diff
@@ -156,8 +156,7 @@ private synchronized static ResourceLimiter initializeResourceLimiter(BigtableOp
 
   private Watchdog watchdog;
   private final BigtableDataClient dataClient;
-
-  private final IBigtableDataClient clientWrapper;
+  private final RequestContext dataRequestContext;
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: Why remove the wrapper? I thought the eventual goal is to migrate to using that interface only

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java [hunk 2]
```diff
@@ -358,6 +360,8 @@ int getRequestCount() {
    *          {@link MutateRowRequest}s will be sent.
    * @param client a {@link BigtableDataClient} object on which to perform RPCs.
    *          RPCs that this object performed.
+   * @param clienWrapper a {@link IBigtableDataClient} object on which to perform RPCs.
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java]: s/clienWrapper/clientWrapper

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkRead.java [hunk 3]
```diff
@@ -56,6 +57,7 @@
 
   /** Constant <code>LOG</code> */
   protected static final Logger LOG = new Logger(BulkRead.class);
+  private static final RequestContext DUMMY_CONTEXT = RequestContext.create("p", "r", "");
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkRead.java]: Why do you need a dummy context? you have a real tableName

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutatorHelper.java [hunk 4]
```diff
@@ -227,7 +222,7 @@ public long getWriteBufferSize() {
   }
 
   private MutateRowsRequest.Entry toEntry(RowMutation rowMutation) {
-    return rowMutation.toBulkProto(requestContext).getEntries(0);
+    return rowMutation.toBulkProto(RequestContext.create("p", "i", "a")).getEntries(0);
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutatorHelper.java]: Why not pass a real name to the constructor and use it here?

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableDataClient.java [hunk 5]
```diff
@@ -36,12 +36,9 @@
  */
 public class BigtableDataClient {
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableDataClient.java]: I'm starting to get lost in the wrappers. Not necessarily in this PR, but we add some doc and the layering

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java (hunk 1 above)
Comment (sduskis) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: I removed the variable in `BigtableSession`.  Creating new `IBigtableDataClient` is cheap, and I added a new throttled `IBigtableDataClient`.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableDataClient.java (hunk 5 above)
Comment (sduskis) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableDataClient.java]: I have a PR in progress to remove this wrapper.

**Comment by sduskis** [comment]
@igorbernstein2, all issues were addressed.  PTAL.

**Pull request merged**